src/OpenTelemetry.Exporter.Geneva/Internal/Tld/TldLogCommon.cs

```diff
@@ -0,0 +1,232 @@
+            result[0] = (char)(firstChar - 32);
+            ++validNameLength;
+        }
+        else
+        {
+            // Not a valid name.
+            return string.Empty;
+        }
+
+        for (var i = 1; i < categoryName.Length; i++)
+        {
+            if (validNameLength == MaxSanitizedEventNameLength)
+            {
+                break;
+            }
+
+            var cur = categoryName[i];
+            if (cur is >= 'a' and <= 'z' or >= 'A' and <= 'Z' or >= '0' and <= '9')
+            {
+                result[validNameLength] = cur;
+                ++validNameLength;
+            }
+        }
+
+        return result.Slice(0, validNameLength).ToString();
+    }
+
+    protected static void OnProcessScopeForIndividualColumns(
+        LogRecordScope scope,
+        SerializationDataForScopes stateDataValue,
+        HashSet<string>? customFields,
+        KeyValuePair<string, object>[] partCFieldsValue,
+        List<KeyValuePair<string, object?>>? envProperties)
+    {
+        Debug.Assert(stateDataValue != null, "state.serializationData was null");
+        Debug.Assert(PartCFields.Value != null, "PartCFields.Value was null");
+
+        var stateData = stateDataValue!;
+        var kvpArrayForPartCFields = PartCFields.Value!;
+
+        List<KeyValuePair<string, object?>>? envPropertiesList = null;
+
+        foreach (var scopeItem in scope)
+        {
+            if (string.IsNullOrEmpty(scopeItem.Key) || scopeItem.Key == "{OriginalFormat}")
+            {
+                continue;
+            }
+
+            if (customFields == null || customFields.Contains(scopeItem.Key))
+            {
+                if (scopeItem.Value != null)
+                {
+                    kvpArrayForPartCFields[stateData.PartCFieldsCountFromState] = new(scopeItem.Key, scopeItem.Value);
+                    stateData.PartCFieldsCountFromState++;
+                }
+            }
+            else
+            {
+                if (stateData.HasEnvProperties == 0)
+                {
+                    stateData.HasEnvProperties = 1;
+
+                    envPropertiesList = EnvProperties.Value ??= [];
+
+                    envPropertiesList.Clear();
+                }
+
+                // TODO: This could lead to unbounded memory usage.
+                envPropertiesList!.Add(new(scopeItem.Key, scopeItem.Value));
+            }
+        }
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (this.isDisposed)
+        {
+            return;
+        }
+
+        if (disposing)
+        {
+            try
+            {
+                // DO NOT Dispose eventBuilder, keyValuePairs, and partCFields as they are static
+                this.serializationData?.Dispose();
+            }
+            catch (Exception ex)
+            {
+                ExporterEventSource.Log.ExporterException("TraceLoggingDynamicCommon Dispose failed.", ex);
```

Comment:
Where does the name "TraceLoggingDynamicCommon" come from?

Reply:
It was a previous rename. I will change it.